app/javascript/utilities/calculateTextAreaHeight.js

```diff
@@ -0,0 +1,103 @@
+const HIDDEN_TEXTAREA_STYLE = `
+min-height:0 !important;
+max-height:none !important;
+height:0 !important;
+visibility:hidden !important;
+overflow:hidden !important;
+position:absolute !important;
+z-index:-1000 !important;
+top:0 !important;
+right:0 !important
```

Comment:
I'm curious: why do we need all these `!important;`s here? Is it because of some crayons-specific thing, or something else? 🤔 It might be nice to leave a comment here so that future folks looking at this file know why we intentionally had to put them there, if that is indeed the case!

Reply:
I'll add a comment in the code as well, but it's because the function that consumes this works by building an invisible copy of an existing DOM element, applying all styles. This lets us calculate the size of the visible text area's content. 

We use all the `!important`s because although we want to inherit _nearly_ every style from the visible element (otherwise the calculated height would be inaccurate), we definitely **don't** want this hidden text area to be visible, or appear causing layout jumps.